packages/cli/src/lib/implementation/config-middleware.unit.test.ts

```diff
@@ -0,0 +1,23 @@
+import { describe, expect, vi } from 'vitest';
+import { configMiddleware } from './config-middleware';
+
+// Mock bundleRequire inside importEsmModule used for fetching config
+vi.mock('bundle-require', async () => {
+  const { CORE_CONFIG_MOCK }: typeof import('@code-pushup/testing-utils') =
+    await vi.importActual('@code-pushup/testing-utils');
+  return {
+    bundleRequire: vi
+      .fn()
+      .mockResolvedValue({ mod: { default: CORE_CONFIG_MOCK } }),
+  };
+});
+
+describe('configMiddleware', () => {
+  it('should load code-pushup.config.(ts|mjs|js)', async () => {
+    // It is indirectly testing if code-pushup.config.(ts|mjs|ts) is used.
+    // The reason for this is a hard to setup test
+    await expect(configMiddleware({})).rejects.toThrow(
+      'No file code-pushup.config',
+    );
+  });
```

Comment:
I think you'd have to create one these files with `memfs`'s `vol.fromJSON`. The reason it's complicated is because there's an (unneccessary, I believe) check with `fs.stat` before `bundle-require` is called (which checks the file exists then anyway), which means both have to be simulated in tests.

But since we already have integration tests set up for the config middleware, why not just add this test case there instead? Something like:

```ts
  it('should load code-pushup.config.(ts|mjs|js) by default', async () => {
    const config = await configMiddleware({});
    expect(config.config).toContain('code-pushup.config');
  });
```

Reply:
It would require to also mock fs in the integration tests etc ... But it is added now and checks against the root config